services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -72,12 +72,58 @@
     @Autowired
     private SubnetService subnetService;
 
+    @RequestMapping(
+            method = GET,
+            value = {"/subnets/{rangeId}"})
+    @DurationStatistics
+    public Integer getUsedIPByRangeId(@PathVariable String rangeId) throws Exception {
+
+        Integer usedIps = null;
+
+        try {
+            RestPreconditionsUtil.verifyParameterNotNullorEmpty(rangeId);
+
+            usedIps = this.subnetService.getUsedIpByRangeId(rangeId);
+        } catch (ParameterNullOrEmptyException e) {
+            //TODO: REST error code
+            logger.error(e.getMessage());
+            throw new Exception(e);
```

Comment:
Is ParameterNullOrEmptyException our own exception with a specific error code? If that is the case, it would be a good idea to throw ParameterNullOrEmptyException directly.

Reply:
Sure